tools/pipelines/test-perf-benchmarks.yml

```diff
@@ -560,131 +332,31 @@ stages:
-              echo "registry=https://registry.npmjs.org" >> ./.npmrc
-              echo "always-auth=false" >> ./.npmrc
-
-              echo "@fluidframework:registry=$(ado-feeds-dev)" >> ./.npmrc
-              echo "@fluid-experimental:registry=$(ado-feeds-dev)" >> ./.npmrc
-              echo "@fluid-tools:registry=$(ado-feeds-dev)" >> ./.npmrc
-              echo "@fluid-internal:registry=$(ado-feeds-dev)" >> ./.npmrc
-              echo "@fluid-private:registry=$(ado-feeds-dev)" >> ./.npmrc
-              echo "@ff-internal:registry=$(ado-feeds-dev)" >> ./.npmrc
-              echo "@microsoft:registry=$(ado-feeds-office)" >> ./.npmrc
-              echo "always-auth=true" >> ./.npmrc
-              cat .npmrc
-
-        # Auth to internal feed
-        - task: npmAuthenticate@0
-          displayName: 'npm authenticate (internal feed)'
-          inputs:
-            workingFile: $(testWorkspace)/.npmrc
-
-        # Install aria-logger
-        - task: Npm@1
-          displayName: 'npm install aria logger'
-          inputs:
-            workingDir: $(testWorkspace)
-            command: 'custom'
-            customCommand: 'install @ff-internal/aria-logger'
-            customRegistry: 'useNpmrc'
-
-        # Install e2e test package
-        - task: Bash@3
-          displayName: Install Test Package
-          inputs:
-            targetType: 'inline'
-            workingDirectory: ${{ variables.testWorkspace }}
-            script: |
-              echo "Install ${{ variables.testPackage }}"
-              TEST_PACKAGE_FILE_PATTERN=${{ variables.escapedTestPackage }}-?.?.?-*.tgz
-              TEST_PACKAGE_PATH_PATTERN=$(Pipeline.Workspace)/client/pack/scoped/$TEST_PACKAGE_FILE_PATTERN
-
-              echo $TEST_PACKAGE_FILE_PATTERN
-              echo $TEST_PACKAGE_PATH_PATTERN
-
-              echo `ls -1 $TEST_PACKAGE_PATH_PATTERN | wc -l`
-              echo `ls $TEST_PACKAGE_PATH_PATTERN`
-
-              if [[ `ls -1 $TEST_PACKAGE_PATH_PATTERN | wc -l` -eq 1 ]]; then
-                TEST_PACKAGE_TGZ=`ls $TEST_PACKAGE_PATH_PATTERN`
-              else
-                ls -1 $TEST_PACKAGE_PATH_PATTERN
-                echo "##vso[task.logissue type=error]Test package '${{ variables.testPackage }}' not found, or there are more then one found"
-                exit -1
-              fi
-
-              npm install $TEST_PACKAGE_TGZ
+        # Download and install package with performance tests
+        - template: templates/include-test-perf-benchmarks-install-package.yml
+          parameters:
+            # Not sure why macro syntax $() for 'testPackage' doesn't work here, maybe because the template is parsed
+            # at "compile time"? The others are defined at the pipeline level so maybe that's why they work...?
+            artifactBuildId: $(artifactBuildId)
+            artifactPipeline: $(artifactPipeline)
+            testPackageName: ${{ variables.testPackage }}
+            installPath: $(testWorkspace)
```

Comment:
This was an interesting bit. Using macro syntax `$(testPackage)` did not work as I thought, probably because the template gets processed at "compile time", before macro syntax expressions are processed, so the template was receiving `$(testPackage)` as a string and its attempt to use it as part of a step name failed due to invalid characters.

Reply:
Outdated comment